### ISSUE TITLE
Getting White screen in MacOS High sierra at random times

### ISSUE THREAD

**Issue by ravening** [mattermost/desktop#994]
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [ ] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**

**Environment**
- Operating System: macOS High Sierra 10.13.3
- Mattermost Desktop App version: 4.2.1 (4.2.1.62)
- Mattermost Server version: 4.2.1 (4.2.1.62)

**Steps to reproduce**

1. Login to MatterMost.
2. Wait for around 10 minutes
3. The main screen turn white
4. The same issue happens multiple times

**Expected behavior**
The screen should not turn blank or white

**Observed behavior**
The screen frequently turns white/blank even after disabling GPU acceleration

**Comment by jby** [comment]
I'm seing this with the following environments as well:

* macOS Mojave 10.14.5
* MM Desktop App: 4.2.1 (4.2.1.62)
* MM Server: 5.12

* macOS Mojave 10.14.5
* Safari webclient
* MM Server: 5.12

* Windows 10 
* MM Desktop 4.3.0
* MM Server: 5.12

The user with Safari on Mac sees websocket errors in the JS Log in Safari.

**Comment by boxed** [comment]
There seems to be an issue with sorting channel names. I have stopped in the debugger where it crashes. This is the line:

```
return a.name.toLowerCase().localeCompare(b.name.toLowerCase(), locale, {
```

Which is called from:

```
}).sort(_channel_utils.sortChannelsByDisplayName.bind(null, locale))[Symbol.iterator](), _step2; !(_iteratorNormalCompletion2 = (_step2 = _iterator2.next()).done); _iteratorNormalCompletion2 = true) {
```

The list of channels contains an entry `{delete_at: 1561361660360}`, which is where it crashes.

**Comment by svelle** [comment]
@rakgenius @jby and @boxed thank you all for the info provided. 
Our devs are currently working on fixing this issue and I'll let you know as soon as there's any updates on this.

**Comment by boxed** [comment]
If you need more data I can probably supply it. We seem to be getting this error once in a while and I can catch this in the debugger :P

**Comment by swood** [comment]
Dear @svelle.
Probably you have any updates by this issue? We are facing the same problem every day and it is so annoying.

**Comment by boxed** [comment]
@swood the workaround for us was to restrict access to the archive room feature so random users didn't do it all the time.

**Comment by svelle** [comment]
Hey all, sorry for not responding earlier. 
I get a lot of github notifications so these sometimes go under.

We've identified the cause of this issue in 5.12 and the issue should be fixed when updating to 5.12.5. 
If you can I'd suggest upgrading to 5.13.2 right away because that has several other performance and quality improvements as well. 

Please let me know if you still experience those issues so we can work on resolving those. 

Also if you get white screens in the future it's most likely due to the webapp failing at some point. So if you run into a white screen when entering a channel or navigating the webapp in general a screenshot or copy of the javascript console is really helpful.

**Comment by swood** [comment]
Thank you, I will try to update my server installation. Now it is 5.12.0.

**Comment by swood** [comment]
@boxed, @svelle.
I have an application console during a day. I have a lot of errors related to WebSocket. So, my question is - how can I disable this setting from the server-side or client-side? I have about 700 active users and I do not want to have a problem with doesn't work technology.
Please check the screenshot via the next link:
https://snag.gy/bWz3m9.jpg

This is a huge problem, I believe all of us could agree with it.

**Comment by svelle** [comment]
Hey @swood, what server version is this happening on?

**Comment by swood** [comment]
Hey, @svelle. As I mentioned before, not it is 5.12.0.
Before my message I have disabled option "Enable Channel Viewed WebSocket Messages". But I did not restart server after, only reload.

I assume this is the main problem of this issue. If you have a hundreds users I should disable WebSocket setting and restart server.

**Comment by svelle** [comment]
You'll need to update the server for the error to disappear. 
The fix should be in 5.12.5 and 5.13.2.

**Comment by devinbinnie** [comment]
Closing as this is a server issue which has been resolved.
